services/github/github-deployments.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      deployments: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestStatus: Joi.object({
+                state: Joi.string(),
+              }),
+            })
+          )
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class GithubDeployments extends GithubAuthV4Service {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'github/deployments',
+      pattern: ':user/:repo/:environment',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub deployments',
+        namedParams: {
+          user: 'badges',
+          repo: 'shields',
+          environment: 'shields-staging',
+        },
+        staticPreview: this.render({
+          state: 'success',
+        }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'state' }
+  }
+
+  static render({ state }) {
+    const { label, message: build_message, color } = renderBuildStatusBadge({
+      label: 'state',
+      status: state,
+    })
+
+    let message = build_message
+
+    // renderBuildStatusBadge() does everything we need except it rewrites the
+    // message for status in 'greenStatuses' to 'passing'. This makes little
+    // sense in the context of a deployment. So rewrite it back to 'success'.
+    if (state == 'success') {
+      message = 'success'
+    }
+
+    return {
+      label,
+      message,
+      color,
+    }
+  }
+
+  async fetch({ user, repo, environment }) {
+    return await this._requestGraphql({
+      query: gql`
+        query($user: String!, $repo: String!, $environment: String!) {
+          repository(owner: $user, name: $repo) {
+            deployments(last: 1, environments: [$environment]) {
+              nodes {
+                latestStatus {
+                  state
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: { user, repo, environment },
+      schema,
+      transformErrors,
+    })
+  }
+
+  transform({ data }) {
+    if (data.repository.deployments.nodes.length == 0) {
+      return { state: 'unknown' }
+    }
+
+    let state = data.repository.deployments.nodes[0].latestStatus.state
+    state = state.toLowerCase()
+
+    if (state == 'in_progress') {
+      return { state: 'in progress' }
+    }
```

Comment:
This (and the lowercasing) feel more like rendering responsibilities than transform responsibilities IMO. Generally speaking those are, respectively, presenting the data to the user in a pretty/friendly way, and extracting the relevant data portions from the raw response